snuba/subscriptions/executor_consumer.py

```diff
@@ -305,6 +295,12 @@ def submit(self, message: Message[KafkaPayload]) -> None:
                 )
             )
         else:
+            # Just commit offset even if we are skipping execution
+            if self.__commit is not None:
+                self.__commit(
```

Comment:
This would commit every message, this puts a lot of load on Kafka for no reason. Commit is an expensive and synchronous operation (it writes on kafka and waits for replication).
You should only commit periodically.

Reply:
This is intentional. We commit every message that is after it is executed so equally this commits every message that is temporarily skipped. Now it may make sense in the future to try and optimise this, but I don't think it is presently an issue due to the number of subscriptions that we have being rather low.

Reply:
you have hundreds of messages per second that means hundreds of commits per second. This is not a good number for Kafka. a handful of commits per second are ok.